listen/map-changes.js

```diff
@@ -22,26 +19,54 @@ var object_owns = Object.prototype.hasOwnProperty;
     }
 */
 
-var mapChangeDescriptors = new WeakMap();
+function TokenChangeDescriptor() {
+	return this;
+}
 
+Object.defineProperties(TokenChangeDescriptor.prototype,{
+    isActive: {
+		value: false,
+		writable: true
+	},
+    _changeListeners: {
+		value: null,
+		writable: true
+	},
+    changeListeners: {
+		get: function() {
+			return this._changeListeners || (this._changeListeners = []);
+		}
+	},
+    _willChangeListeners: {
+		value: null,
+		writable: true
+	},
+    willChangeListeners: {
+		get: function() {
+			return this._willChangeListeners || (this._willChangeListeners = []);
+		}
+	}
+});
+
+MapChanges.prototype._mapChangeDescriptors = null;
```

Comment:
defineProperties is excessive for values. Note that configurable and enumerable are false by default, and simple assignment is enumerable true by default.